### DISCUSSION TITLE
Token for protected SID API's

### DISCUSSION THREAD

**Discussion by agkoutis** [simpleidserver/SimpleIdServer#642] [Q&A]
Hello!

Is there any internal process for creating a token and interacting with a protected API from SID? or do I have to create it manually? (make SID client and self-request token)

Thanks

**Comment by simpleidserver** [comment]
Currently, we lack a client that can be utilized to interact with the various REST APIs exposed by SimpleIdServer.

Nevertheless, in the upcoming release of SimpleIdServer, the Swagger endpoint will be made accessible. 
You can employ a tool such as NSwagStudio to generate the client. 
Refer to this link for guidance: https://learn.microsoft.com/en-us/aspnet/core/tutorials/getting-started-with-nswag?view=aspnetcore-8.0&tabs=visual-studio#generate-code-with-nswagstudio

The master branch incorporates the modifications to support the Swagger library.

**Comment by simpleidserver** [comment]
@agkoutis : By the way, I forgot to notify you about the changes made on the 'master' branch.

All the LINQ expressions have been moved to the UserRepository. It is now easier for a developer to override different methods, such as 'GetBySubject'.

**Reply by agkoutis** [reply]
Thank you for the notification!

I will check it as soon as possible
